coalib/io/Directory.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+from coalib.parsing.Globbing import relative_recursive_glob
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('path')
+class Directory:
+    def __init__(self, path):
+        self._path = os.path.abspath(path)
+        self._parent = os.path.abspath(os.path.dirname(self._path))
```

Comment:
Is it possible to maybe add a timestamp here as well? Since `Directory`s will be cached too.